code/game/gamemodes/changeling/powers/panacea.dm

```diff
@@ -8,8 +8,19 @@
 
 //Heals the things that the other regenerative abilities don't.
 /obj/effect/proc_holder/changeling/panacea/sting_action(mob/user)
-
 	user << "<span class='notice'>We cleanse impurities from our form.</span>"
+
+	var/obj/item/organ/internal/body_egg/egg = user.getorgan(/obj/item/organ/internal/body_egg)
+	if(egg)
+		egg.Remove(user)
+		user.visible_message("<span class='danger'>[user] vomits up [egg]!</span>", "<span class='userdanger'>[user] vomits up [egg]!</span>")
+		playsound(user.loc, 'sound/effects/splat.ogg', 50, 1)
+
+		var/turf/location = user.loc
+		if(istype(location, /turf/simulated))
+			location.add_vomit_floor(user, 1)
+		egg.loc = location
+

```

Comment:
This will conflict with my own fix for this (#11883), but your idea is probably better than mine.